crates/subspace-networking/src/create.rs

```diff
@@ -224,6 +225,8 @@ pub struct Config<ProviderStorage> {
     pub special_target_connections: u32,
     /// Addresses to bootstrap Kademlia network
     pub bootstrap_addresses: Vec<Multiaddr>,
+    /// Optionally overrides the default Kademlia server mode.
+    pub kademlia_mode_override: Option<Option<Mode>>,
```

Comment:
Why not just `Option<Mode>` and default it to `Some(Mode::Client)`?

Reply:
Better yet, just `Mode`

Reply:
Why do we need automatic mode? I thought the whole point was to make it explicit

Reply:
Initially, I set the `Server` mode to upgrade the libp2p with preserving the current behavior. However, It would be great to use the automatic mode, otherwise, we should add basically the same code as they did. On the other hand, when we know that we must not get to the routing table (as for temporary clients now or for nodes in the future) it's convenient to set `Client` mode explicitly. I believe that having all options available is good.

Reply:
It still looks very strange. If we remove outer `Option` we just don't set anything on `None` (libp2p default) and if it is `Some` then we do set it to whatever was specified. I don't think `Some(None)` is useful.

Reply:
I change the semantics then.